routers/routes/routes.go

```diff
@@ -135,6 +135,9 @@ func NewMacaron() *macaron.Macaron {
 	if setting.Protocol == setting.FCGI {
 		m.SetURLPrefix(setting.AppSubURL)
 	}
+	if setting.EnableCORS {
```

Comment:
It is maybe a stupid question but why we would need CORS on all the paths ?
Furthermore if it is needed, this my get in double with L961 (recently introduced here in #7967).

Reply:
Because resources will be a different domain from the main page. We needs suitable CORS on resources requests.

Reply:
From what I understand the goal is to serve file from another kind of server and cors is not needed for js, img, ...

Reply:
The web fonts need CORS headers. But iirc they are only needed on the response of the cdn.

Reply:
if the static URL domain is different from main domain, we should not follow `setting.EnableCORS`

Reply:
@lunny and if static URL domain is equal to main domain it is not needed anymore. We should remove this part.

Reply:
Done in 743b2f769600a983f1698cdb9338a617d9136b98